driver-core/src/main/com/mongodb/internal/operation/AsyncCommandBatchCursor.java

```diff
@@ -90,6 +91,8 @@ class AsyncCommandBatchCursor<T> implements AsyncAggregateResponseBatchCursor<T>
         this.maxWireVersion = connectionDescription.getMaxWireVersion();
         this.firstBatchEmpty = commandCursorResult.getResults().isEmpty();
 
+        connectionSource.getOperationContext().getTimeoutContext().setMaxTimeSupplier(() -> maxTimeMS);
```

Comment:
Not sure `maxTimeMS` is needed in this class (or the sync equivalent).  Setting of MaxTimeSupplier could be done in a similar way as `addMaxTimeMSToNonTailableCursor` is used.

Reply:
Given that `maxTime` is closely linked to the cursor's behavior, maintaining its setting within proximity to the related logic, i believe, improves clarity. Setting maxTime outside of the `CommandBatch `cursor could lead to confusion, as it might not be immediately apparent that there is `maxTime` which is configured externally from the cursor class.